File: app/api/v1/schemas/stats.py
"""
统计相关 Schema
"""

from datetime import datetime
from typing import Optional, List, Dict
from pydantic import BaseModel, Field


class DiskUsage(BaseModel):
    """磁盘使用情况"""
    total: int = Field(..., description="总空间(字节)")
    used: int = Field(..., description="已用空间(字节)")
    free: int = Field(..., description="剩余空间(字节)")
    percent: float = Field(..., description="使用百分比")


class WorkflowStatus(BaseModel):
    """工作流状态"""
    name: str = Field(..., description="工作流名称")
    display_name: str = Field(..., description="显示名称")
    status: str = Field(..., description="状态: running/stopped")
    last_run: Optional[datetime] = Field(None, description="上次执行时间")
    next_run: Optional[datetime] = Field(None, description="下次执行时间")
    interval: str = Field(..., description="执行间隔")


class RecentActivity(BaseModel):
    """近期活动"""
    id: int = Field(..., description="记录ID")
    content_id: str = Field(..., description="番号")
    action: str = Field(..., description="操作类型")
    status: str = Field(..., description="状态")
    message: Optional[str] = Field(None, description="消息")
    created_at: datetime = Field(..., description="时间")


class TaskStatusCount(BaseModel):
    """任务状态统计"""
    status: str = Field(..., description="状态")
    count: int = Field(..., description="数量")


class StatsResponse(BaseModel):
    """统计信息响应"""
    # 任务统计
    total_tasks: int = Field(..., description="总任务数")
    running_tasks: int = Field(..., description="进行中")
    pending_tasks: int = Field(..., description="等待中")
    completed_tasks: int = Field(..., description="已完成")
    failed_tasks: int = Field(..., description="失败")
    timeout_tasks: int = Field(..., description="超时")
    
    # 今日统计
    completed_today: int = Field(..., description="今日完成")
    created_today: int = Field(..., description="今日创建")
    
    # 状态分布
    status_distribution: List[TaskStatusCount] = Field(..., description="状态分布")
    
    # 磁盘使用
    disk_usage: Optional[DiskUsage] = Field(None, description="磁盘使用")
    
    # 工作流状态
    workflow_status: List[WorkflowStatus] = Field(..., description="工作流状态")


class RecentStatsResponse(BaseModel):
    """近期活动响应"""
    activities: List[RecentActivity] = Field(..., description="活动列表")
    total: int = Field(..., description="总数")

File: web/src/api/stats.ts
import apiClient from './client'

export interface DiskUsage {
  total: number
  used: number
  free: number
  percent: number
}

export interface WorkflowStatus {
  name: string
  display_name: string
  status: string
  last_run?: string
  next_run?: string
  interval: string
}

export interface TaskStatusCount {
  status: string
  count: number
}

export interface StatsResponse {
  total_tasks: number
  running_tasks: number
  pending_tasks: number
  completed_tasks: number
  failed_tasks: number
  timeout_tasks: number
  completed_today: number
  created_today: number
  status_distribution: TaskStatusCount[]
  disk_usage: DiskUsage | null
  workflow_status: WorkflowStatus[]
}

export interface RecentActivity {
  id: number
  content_id: string
  action: string
  status: string
  message: string | null
  created_at: string
}

export interface RecentStatsResponse {
  activities: RecentActivity[]
  total: number
}

export const statsApi = {
  getStats: (): Promise<StatsResponse> =>
    apiClient.get('/stats').then(res => res.data),
  
  getRecentActivities: (limit: number = 10): Promise<RecentStatsResponse> =>
    apiClient.get('/stats/recent', { params: { limit } }).then(res => res.data),
}
